test/libsolidity/syntaxTests/inheritance/dataLocation/memory_calldata_public.sol

```diff
@@ -0,0 +1,16 @@
+abstract contract A {
+    function f(uint256[1] memory a) public virtual returns (uint256);
+    function test() external returns (uint) {
+        uint[1] memory t;
+        t[0] = 7;
+        return f(t);
+    }
+}
+
+contract B is A {
+    function f(uint256[1] calldata a) public override returns (uint256) {
```

Comment:
I think we should also have the other direction (`calldata` -> `memory`) covered.